SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestHandoverReplication/SpatialTestHandoverReplication.cpp

```diff
@@ -0,0 +1,210 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestHandoverReplication.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Kismet/GameplayStatics.h"
+#include "LoadBalancing/LayeredLBStrategy.h"
+
+#include "DynamicReplicationHandoverCube.h"
+#include "SpatialFunctionalTestFlowController.h"
+
+/**
+ * This tests that an Actor's bReplicate flag is properly handed over when dynamically set.
+ * Tests UNR-4441
+ * This test contains 4 Server and 2 Client workers.
+ *
+ * The flow is as follows:
+ * - Setup:
+ *	- Server 1 spawns a HandoverCube with bReplicates set to false inside its authority area. (ADynamicReplicationHandoverCube)
+ *	- The bReplicates flag is set to true after the end of the Actor's initialization.
+ *  - All servers set a reference to the HandoverCube and reset their local copy of the LocationIndex and AuthorityCheckIndex.
+ * - Test:
+ *	- At this stage, Server 1 should have authority over the HandoverCube.
+ *  - The HandoverCube is moved in the authority area of Server 2.
+ *  - At this stage, Server 2 should have authority over the HandoverCube.
+ *  - Server 2 acquires a lock on the HandoverCube and moves it into the authority area of Server 4.
```

Comment:
It's not a deal breaker but is there a reason that the cube moves from 1,2,4,3 rather than sequentially 1,2,3,4?

If this can be achieved by an quick re-naming I think that would be worthwhile, as it being our of sequential order suggests a complexity to the test that, as far as I can tell, does not exist.